imap-proto/src/parser/rfc5256.rs

```diff
@@ -0,0 +1,27 @@
+//!
+//! https://tools.ietf.org/html/rfc5256
+//!
+//! SORT extension
+//!
+
+use nom::{
+    bytes::streaming::{tag, tag_no_case},
+    combinator::{map, opt},
+    multi::many0,
+    sequence::{preceded, terminated},
+    IResult,
+};
+
+use crate::{parser::core::number, types::MailboxDatum};
+
+pub(crate) fn mailbox_data_sort(i: &[u8]) -> IResult<&[u8], MailboxDatum> {
```

Comment:
Can you add a docstring here with the syntax from the relevant RFC, and maybe a little context on what this is/how it's used? (There are many existing parsers that have stuff like this, see #116 for some examples.)